lib/convection/model/template.rb

```diff
@@ -183,9 +197,122 @@ def diff(other, stack_ = nil)
       end
 
       def to_json(stack_ = nil, pretty = false)
+        rendered_stack = render(stack_)
+        validate(rendered_stack)
+
         return JSON.generate(render(stack_)) unless pretty
```

Comment:
Instead of rendering the stack multiple lines re-use the `rendered_stack` variable here and on the line below.